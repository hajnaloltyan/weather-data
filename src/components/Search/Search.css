.header-border {
  margin: 2% 5% 4%;
}

.wrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #4369b2;
  margin: 1% 8%;
}

.pageTitle {
  font-size: 22px;
  font-weight: 700;
  color: #fff;
}

.home {
  font-size: 16px;
  font-weight: 700;
  background-color: #4369b2;
}

.searchContainer {
  font-size: 16px;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 2% 20%;
  position: relative;
  margin-top: 8%;
}

.searchInput {
  width: 180px;
  padding: 2px 10px;
  border: 0;
  border-bottom: 1px solid #000;
  outline: none;
  margin-top: 10px;
  font-size: 14px;
}

.searchResults {
  background-color: #fff;
  width: 180px;
  z-index: 1;
  position: absolute;
  top: 100%;
  font-size: 14px;
}

.searchItem {
  padding: 5px 10px;
  cursor: pointer;
  width: 180px;
  text-align: left;
  border: 0;
}

.searchItem:hover {
  background-color: #ec4c8a;
}
