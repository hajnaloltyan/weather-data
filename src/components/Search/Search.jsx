import { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, NavLink } from 'react-router-dom';
import { BsFillHouseFill } from 'react-icons/bs';
import { fetchPlaceNames, AllSearches, setLocation } from '../../redux/search/searchSlice';
import { fetchWeatherData } from '../../redux/weatherData/weatherDataSlice';
import './Search.css';

const Search = () => {
  const [query, setQuery] = useState('');
  const dispatch = useDispatch();

  const placeNames = useSelector(AllSearches);
  useEffect(() => {
    if (query.trim() !== '' && query.length > 2) {
      dispatch(fetchPlaceNames(query));
    }
  }, [dispatch, query]);

  const handleInputChange = (event) => {
    const inputValue = event.target.value;
    setQuery(inputValue);
  };

  const handleSelectLocation = (name, lat, lon) => {
    dispatch(fetchWeatherData({ lat, lon }));
    setQuery('');
    dispatch(setLocation(name));
  };

  const renderPlaceNames = (places) => {
    if (places.length === 0 || query.trim() === '') {
      return null;
    }

    return (
      <ul className="searchResults">
        {places.map(({
          id, lat, lon, name, state, country,
        }) => (
          <li key={id}>
            <Link to="details/options">
              <button
                type="button"
                className="searchItem"
                onClick={() => handleSelectLocation(name, lat, lon)}
              >
                {name}
                ,
                {' '}
                {state}
                ,
                {' '}
                {country}
              </button>
            </Link>
          </li>
        ))}
      </ul>
    );
  };

  return (
    <div className="header-border">
      <div className="wrapper">
        <h1 className="pageTitle">Weather Data</h1>
        <nav className="home">
          <NavLink to="/">
            <BsFillHouseFill size={25} />
          </NavLink>
        </nav>
      </div>
      <div className="searchContainer">
        <p>Please select a location:</p>
        <input
          type="text"
          value={query}
          onChange={handleInputChange}
          placeholder="Search for a place..."
          className="searchInput"
        />
        {renderPlaceNames(placeNames)}
      </div>
    </div>
  );
};

export default Search;
