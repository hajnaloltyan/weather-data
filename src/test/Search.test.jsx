import {
  expect, test,
} from 'vitest';
import Search from '../components/Search/Search';

test('Search component renders correctly', () => {
  const tree = <Search />;

  expect(tree).toMatchSnapshot();
});
