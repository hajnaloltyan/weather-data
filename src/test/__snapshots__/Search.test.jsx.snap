// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search component renders correctly 1`] = `<Search />`;
